src/python/pants/backend/docgen/tasks/markdown_to_html_utils.py

```diff
@@ -66,57 +72,91 @@ def choose_include_text(s, params, source_path):
   :param params: string like "start-at=foo&end-at=bar"
   :param source_path: path to source .md. Useful in error messages
   """
-  lines = s.splitlines()
-  start_after = None
-  start_at = None
-  end_before = None
-  end_at = None
+  source_lines = s.splitlines()
+  params_dict = {}
 
   for term in params.split("&"):
     if '=' in term:
       param, value = [p.strip() for p in term.split('=', 1)]
     else:
       param, value = term.strip(), ''
     if not param: continue
-    if param == "start-after":
-      start_after = value
-    elif param == "start-at":
-      start_at = value
-    elif param == "end-before":
-      end_before = value
-    elif param == "end-at":
-      end_at = value
+    if param not in INCLUDE_RECOGNIZED_PARAMS:
+      raise TaskError('Invalid include directive "{0}" in params "{1}"'
```

Comment:
Now that you have a set of valid parameters, could render it in the error message.

Reply:
Absolutely, will add that to the error message.